DirectXTex/BC.cpp

```diff
@@ -35,9 +35,9 @@ namespace
     //-------------------------------------------------------------------------------------
     inline void Decode565(_Out_ HDRColorA *pColor, _In_ const uint16_t w565)
     {
-        pColor->r = (float)((w565 >> 11) & 31) * (1.0f / 31.0f);
-        pColor->g = (float)((w565 >> 5) & 63) * (1.0f / 63.0f);
-        pColor->b = (float)((w565 >> 0) & 31) * (1.0f / 31.0f);
+        pColor->r = static_cast<float>((w565 >> 11) & 31) * (1.0f / 31.0f);
```

Comment:
``-Wold-school-casts`` in clang found a lot of these lingering in the code, so I fixed them.